script.js

```diff
@@ -1,4 +1,112 @@
+var die1;
+var die2;
+var player1Score = [];
+var player1GlobalScore = 0;
+var player2Score = [];
+var player2GlobalScore = 0;
+var player = 1;
+var mode = "dice roll";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var outputMessage;
+  while (player == 1) {
+    if (mode == "dice roll") {
+      var diceResult = roll2Dice();
+      mode = "choose dice order";
+      outputMessage = `Welcome Player 1. <br>
+     ${diceResult} <br>Choose the order of the dice.`;
+    } else if (mode == "choose dice order") {
+      var player1Number = concatDiceResult(input);
+      player1Score.push("" + player1Number);
+
+      // sum of player 1's score thus far
+      var index = 0;
+      var player1CurrentScore = 0;
+      while (index < player1Score.length) {
+        var currentLoopScore = player1Score[index];
+        player1CurrentScore =
+          Number(player1CurrentScore) + Number(currentLoopScore);
+        index = index + 1;
+      }
+      player1GlobalScore = player1CurrentScore;
+      player = 2;
+      mode = "dice roll";
+      var gameLeader = decideLeader();
+      outputMessage = `Player 1, you chose Dice ${input} first. <br>Your number is ${player1Number}. <br>${gameLeader}<br>It's Player 2's turn.`;
+    }
+    return outputMessage;
+  }
+  while (player == 2) {
+    if (mode == "dice roll") {
+      var diceResult = roll2Dice();
+      mode = "choose dice order";
+      outputMessage = `Welcome Player 2. <br>
+     ${diceResult} <br>Choose the order of the dice.`;
+    } else if (mode == "choose dice order") {
+      var player2Number = concatDiceResult(input);
+      player2Score.push("" + player2Number);
+      console.log("player 2 score " + player2Score);
+
+      // sum of player 2's score thus far
+      var index = 0;
+      var player2CurrentScore = 0;
+      while (index < player2Score.length) {
+        var currentLoopScore = player2Score[index];
+        player2CurrentScore =
+          Number(player2CurrentScore) + Number(currentLoopScore);
+        index = index + 1;
+      }
+      player2GlobalScore = player2CurrentScore;
+      mode = "dice roll";
+      player = 1;
```

Comment:
This whole code seems to be the same as above. This is a big sign that this bunch of code could be abstracted into a separate function and re-used. e.g. handleDiceRoll(player)
within this function, you could create other helper functions such as one to tabulate the total score of a player and updating the global variable